drivers/spi/spi_mcux_flexcomm.c

```diff
@@ -771,17 +806,23 @@ static int spi_mcux_release(const struct device *dev,
 {
 	struct spi_mcux_data *data = dev->data;
 
+	pm_policy_device_power_lock_get(data->dev);
+
 	spi_context_unlock_unconditionally(&data->ctx);
 
+	pm_policy_device_power_lock_put(dev);
+
 	return 0;
 }
 
-static int spi_mcux_init(const struct device *dev)
+static int spi_mcux_init_common(const struct device *dev)
 {
 	const struct spi_mcux_config *config = dev->config;
 	struct spi_mcux_data *data = dev->data;
 	int err = 0;
 
+	pm_policy_device_power_lock_get(dev);
```

Comment:
it looks like this function also is not changing any SPI hardware state, so dont know if the get and put is necessary in this function either